ProjFS.Mac/PrjFSKext/AssociativeArrayTests/AssociativeArrayTests.mm

```diff
@@ -0,0 +1,135 @@
+#import <XCTest/XCTest.h>
+#include <list>
+#import "../PrjFSKext/AssociativeArray.hpp"
+
+@interface AssociativeArrayTests : XCTestCase
+
+@end
+
+static AssociativeArrayTests* s_test;
+static bool s_failNonblockingAlloc = false;
+
+struct TestArrayElement
+{
+    uint32_t key;
+    uint32_t value;
+};
+
+static bool TestArrayElementKeyEquals(const uint32_t& key, const TestArrayElement& item)
+{
+    return item.key == key;
+}
+
+struct MemoryHeader;
+typedef std::list<MemoryHeader*> AllocationList;
```

Comment:
@pmj I really like the groundwork you've set for unit tests here!

Looking at AllocationList, it seems like we could verify how many memory blocks where allocated and in what sizes.  I.e. really have fun testing out any growing scenarios.  Is that correct?

Reply:
Yes, definitely. This was very much a minimal approach illustrating what's possible, as there was a good chance it needed rewriting or would be thrown out.